### PR TITLE
feat(cli): require --confirm for bucket delete and document cascade deletion

### PR DESCRIPTION
**Pull request by sorenbs**
## Purpose of change

Bucket deletion now cascades (the Management API force-deletes all stored objects and revokes all access keys), so the CLI must guard against accidental data loss with the same `--confirm <id>` gate that `database remove` already uses.

## What's changed and why

- **`controllers/bucket.ts`**: `runBucketDelete` now accepts a `BucketDeleteFlags` parameter with `confirm?: string`. Before calling the provider it checks `flags.confirm !== id` and throws `CONFIRMATION_REQUIRED` (exit 2) with `meta: { expectedConfirm, receivedConfirm }` — matching the `database remove` error shape exactly.
- **`commands/bucket/index.ts`**: `createBucketDeleteCommand` wires a `--confirm <bucket-id>` option and passes it through to the controller.
- **`tests/bucket.test.ts`**: Updated the two existing delete tests to include `--confirm`; added tests for missing confirm (null `receivedConfirm`) and mismatched confirm.
- **`docs/product/command-spec.md`**: `bucket delete` heading now shows `--confirm <bucketId>`; behavior section replaces the empty-bucket limitation note with cascade semantics and the confirmation requirement.
- **`docs/product/resource-model.md`**: Object Store section updated to describe cascade deletion and the `--confirm` guard.

## Testing notes / Before-after

**Before**: `prisma-cli bucket delete bkt_123` succeeded immediately.
**After**: same command exits 2 with `CONFIRMATION_REQUIRED`; must pass `--confirm bkt_123`.

Verify manually:
1. `prisma-cli bucket delete bkt_123` → exit 2, error `CONFIRMATION_REQUIRED`
2. `prisma-cli bucket delete bkt_123 --confirm bkt_456` → exit 2, `receivedConfirm: "bkt_456"`
3. `prisma-cli bucket delete bkt_123 --confirm bkt_123` → exit 0, success

**Notes for reviewers:**
- The cascade behavior (`force=true` on the Tigris partner API) is gated on a pdp-control-plane PR landing first. The CLI change is safe to merge independently — it guards deletion regardless of whether the platform side has landed.
- Breaking change: any script passing `bucket delete <id>` without `--confirm` will now exit 2. This is intentional and matches `database remove` precedent.
- `bucket key delete` does **not** get a `--confirm` flag — keys are individually revocable and don't cascade to other resources, so the guard would be disproportionate.

### CODE REVIEW

File: packages/cli/tests/bucket.test.ts [hunk 1]
```diff
@@ -217,11 +217,57 @@ describe("bucket commands", () => {
...
+    expect(wrongPayload).toMatchObject({
+      ok: false,
+      command: "bucket.delete",
+      error: {
+        code: "CONFIRMATION_REQUIRED",
+        domain: "bucket",
+        meta: { expectedConfirm: "bkt_123", receivedConfirm: "bkt_456" },
+      },
+    });
+  });
```
[the remark this answers is not in the document]
Comment (sorenbs) [reply packages/cli/tests/bucket.test.ts:286]: Fixed in 7e15a66: both rejected-confirmation tests now run `bucket list` after the failure and assert bkt_123 is still present, so a regression that deletes despite CONFIRMATION_REQUIRED can't pass.
